### PR TITLE
RPC: Mark storage methods as `blocking` 

### PR DESCRIPTION
**Pull request by lexnv**
`jsonrpsee` supports `sync`, `async` and `blocking` methods.

The storage functions may result in long-time calls, which would block the RPC's
server task. As a result, the RPC fails to respond to customer requests until
the log time call is resolved. Ensure those tasks are spawned as blocking.

One such example is the RPC server being blocked by many DB operations
(intensive sync I/O). This would mean that the RPC wouldn't respond to any
WebSocket `ping` frames, causing the clients that implement a variety of
`ping`-`pong` packets to drop the connection as unhealthy.
 
Part of #11401.

### CODE REVIEW

File: client/rpc/src/state/mod.rs [hunk 1]
```diff
@@ -60,29 +60,29 @@ where
 	Client: Send + Sync + 'static,
 {
 	/// Call runtime method at given block.
-	async fn call(
+	fn call(
```
Comment (niklasad1) [inline client/rpc/src/state/mod.rs:63]: the trait def above doesn't have to be async anymore :point_up: ?

File: client/rpc-api/src/chain/mod.rs [hunk 2]
```diff
@@ -27,11 +27,11 @@ pub mod error;
 pub trait ChainApi<Number, Hash, Header, SignedBlock> {
 	/// Get header.
 	#[method(name = "chain_getHeader")]
-	async fn header(&self, hash: Option<Hash>) -> RpcResult<Option<Header>>;
+	fn header(&self, hash: Option<Hash>) -> RpcResult<Option<Header>>;
 
 	/// Get header and body of a relay chain block.
 	#[method(name = "chain_getBlock")]
-	async fn block(&self, hash: Option<Hash>) -> RpcResult<Option<SignedBlock>>;
+	fn block(&self, hash: Option<Hash>) -> RpcResult<Option<SignedBlock>>;
```
Comment (niklasad1) [inline client/rpc-api/src/chain/mod.rs:34]: these should probably be blocking too?

**Comment by niklasad1** [comment]
To elaborate these calls might fuck up the async scheduler instead with the `blocking` attribute jsonrpsee will call `tokio::spawn_blocking` instead to be spawned on the `blocking threadpool` instead.

Previously we just ported over `FutureResult from jsonrpc` to `async`.

**Review by niklasad1 — APPROVED** [review]
LGTM

File: client/rpc/src/state/state_full.rs [hunk 3]
```diff
@@ -599,18 +599,20 @@ where
...
-		.await
+
+		keys.into_iter()
+			.map(move |key| {
+				let res = client
+					.clone()
+					.child_storage(&BlockId::Hash(block), &child_info, &key)
+					.map_err(client_err);
+
+				res
```
Comment (bkchr) [inline client/rpc/src/state/state_full.rs:610]:
```suggestion
				client
					.clone()
					.child_storage(&BlockId::Hash(block), &child_info, &key)
					.map_err(client_err)
```

**Review by bkchr — APPROVED** [review]

**Comment by lexnv** [comment]
Test failure

```
failures:
    tests::subscribe_and_unsubscribe_to_justifications
```

This happens on other PRs (https://github.com/paritytech/substrate/pull/11461), which must be unrelated to this PR.

**Review by dvdplm — APPROVED** [review]
Nice, thank you for taking care of this. :)

**Comment by niklasad1** [comment]
bot merge

**Pull request merged**